lib/msal-browser/test/interaction_client/RedirectClient.spec.ts

```diff
@@ -1827,6 +1830,461 @@ describe("RedirectClient", () => {
             });
             redirectClient.handleRedirectPromise("", rootMeasurement);
         });
+
+        it("retries on invalid_grant error when onRedirectNavigate is set in config", async () => {
+            let pca = new PublicClientApplication({
+                auth: {
+                    clientId: TEST_CONFIG.MSAL_CLIENT_ID,
+                    onRedirectNavigate: (url: string) => { },
+                },
+            });
+
+            //PCA implementation moved to controller
+            pca = (pca as any).controller;
+
+            // @ts-ignore
+            redirectClient = new RedirectClient(
+                // @ts-ignore
+                pca.config,
+                // @ts-ignore
+                pca.browserStorage,
+                // @ts-ignore
+                pca.browserCrypto,
+                // @ts-ignore
+                pca.logger,
+                // @ts-ignore
+                pca.eventHandler,
+                // @ts-ignore
+                pca.navigationClient,
+                // @ts-ignore
+                pca.performanceClient,
+                // @ts-ignore
+                pca.nativeInternalStorage,
+                undefined,
+                TEST_CONFIG.CORRELATION_ID
+            );
+
+            const stateString = TEST_STATE_VALUES.TEST_STATE_REDIRECT;
+            const browserCrypto = new CryptoOps(new Logger({}));
+            const stateId = ProtocolUtils.parseRequestState(
+                browserCrypto,
+                stateString
+            ).libraryState.id;
+
+            window.sessionStorage.setItem(
+                `${Constants.CACHE_PREFIX}.${TEST_CONFIG.MSAL_CLIENT_ID}.${TemporaryCacheKeys.ORIGIN_URI}`,
```

Comment:
These cache items are duplicated throughout this test suite. I'm considering a separate clean-up PR for this, as time allows.